### ISSUE TITLE
RC Test: Gravity and Orbits 1.4.0-rc.1 PhET-iO and PhET Brand

### ISSUE THREAD

**Issue by samreid** [phetsims/QA#595]
@ariel-phet @KatieWoe @arouinfar @kathy-phet, gravity-and-orbits/1.4.0-rc.1 is ready for RC testing. We would like to publish by Feb 14, 2021.
Document issues in https://github.com/phetsims/gravity-and-orbits/issues and link to this issue.

This is an RC Lite test--like a dev test since it hasn't been tested for a while, but with recorded SHAs in case things go smoothly, and so that we don't need to duplicate testing when we move forward to RC.

Also, to clarify, this is a "placeholder" issue--when QA team is ready for testing, let me know and I'll publish the RC version.  UPDATE: It's been published and links provided.

<details>
<summary><b>General RC Test</b></summary>

<h3>What to Test</h3>

- Click every single button.
- Test all possible forms of input.
  - Test all mouse/trackpad inputs.
  - Test all touchscreen inputs.
- If there is sound, make sure it works.
- Make sure you can't lose anything.
- Play with the sim normally.
- Try to break the sim.
- Test all query parameters on all platforms. (See [QA Book](https://github.com/phetsims/QA/blob/master/documentation/qa-book.md)
for a list of query parameters.)
- Download HTML on Chrome and iOS.
- Make sure the iFrame version of the simulation is working as intended on all platforms.
- Make sure the XHTML version of the simulation is working as intended on all platforms.
- Complete the test matrix.
- Don't forget to make sure the sim works with Legends of Learning.
- Test the Game Up harness on at least one platform.
- Check [this](https://docs.google.com/spreadsheets/d/1umIAmhn89WN1nzcHKhYJcv-n3Oj6ps1wITc-CjWYytE/edit#gid=0) LoL
spreadsheet and notify AR or AM if it not there.
- If this is rc.1 please do a memory test.
- When making an issue, check to see if it was in a previously published version
- Try to include version numbers for browsers
- If there is a console available, check for errors and include them in the Problem Description.
- As an RC begins and ends, check the sim repo. If there is a maintenance issue, check it and notify developers if
there is a problem.

<h3>Focus and Special Instructions</h3>

[Provide further instructions here. If you have any further tests you want done or specific platforms you want tested,
list them here. Any behaviors you want QA to pay special attention to should be listed here.]

<h3>Issues to Verify</h3>

- [x] [User reported orbit inconsistencies between speeds](https://github.com/phetsims/gravity-and-orbits/issues/373)
- [x] [Long strings leave artifacts behind on To Scale Screen](https://github.com/phetsims/gravity-and-orbits/issues/358)

These issues should have the "status:ready-for-qa" label. Check these issues off and close them if they are fixed.
Otherwise, post a comment in the issue saying that it wasn't fixed and link back to this issue. If the label is
"status:ready-for-review" or "status:fixed-pending-deploy" then assign back to the developer when done, even if fixed.

<h3>Link(s)</h3>

Links will be provided on publication.
- **[Simulation](https://phet-dev.colorado.edu/html/gravity-and-orbits/1.4.0-rc.1/phet/gravity-and-orbits_all_phet.html)**
- **[iFrame](https://phet-dev.colorado.edu/html/gravity-and-orbits/1.4.0-rc.1/phet/gravity-and-orbits_all_iframe_phet.html)**
- **[XHTML](https://phet-dev.colorado.edu/html/gravity-and-orbits/1.4.0-rc.1/phet/xhtml/gravity-and-orbits_all.xhtml)**
- **[Test Matrix](https://docs.google.com/spreadsheets/d/1U61MmzaK2YR1MvtvBo1vYZ4zkEpwNroxhbtdhOgNEPs/edit#gid=2423424)**
- **[Legends of Learning Harness](https://developers.legendsoflearning.com/public-harness/index.html?startPayload=%7B%22languageCode%22%3A%22en%22%7D)**

<hr>

</details>

<details>
<summary><b>PhET-iO RC Test</b></summary>

<h3>What to Test</h3>

- Make sure that public files do not have password protection. Use a private browser for this.
- Make sure that private files do have password protection. Use a private browser for this.
- Make sure standalone sim is working properly.
- Make sure the wrapper index is working properly.
- Make sure each wrapper is working properly.
- Launch the simulation in Studio with ?stringTest=xss and make sure the sim doesn't navigate to youtube
- For newer PhET-iO wrapper indices, save the "basic example of a functional wrapper" as a .html file and open it. Make
sure the simulation loads without crashing or throwing errors.
- For an update or maintenance release please check the backwards compatibility of the playback wrapper.
[Here's the link to the previous wrapper.](link)
- Load the login wrapper just to make sure it works. Do so by adding this link from the sim deployed root:
  ```
  /wrappers/login/?wrapper=record&validationRule=validateDigits&&numberOfDigits=5&promptText=ENTER_A_5_DIGIT_NUMBER
  ```
  Further instructions in QA Book
- Conduct a recording test to Metacog, further instructions in the QA Book. Do this for iPadOS + Safari and one other random platform.
- Conduct a memory test on the stand alone sim wrapper (rc.1).
- Test one platform combination with `?phetioDebug` on the Studio and State wrapper.

<h3>Focus and Special Instructions</h3>

[Provide further instructions here. If you have any further tests you want done or specific platforms you want tested,
list them here. Any behaviors you want QA to pay special attention to should be listed here. This includes if any substantial,
untested work has been implemented in PhET-iO wrappers or in the PhET-iO Engine since last PhET-iO test.]

<h3>Link(s)</h3>

- **[Wrapper Index](https://phet-dev.colorado.edu/html/gravity-and-orbits/1.4.0-rc.1/phet-io)**
- **[Test Matrix](https://docs.google.com/spreadsheets/d/1MiLr45_vxmzBlu88343uG3OV-_Pb7qRBAMRTIZKVPhE/edit#gid=1834502962)**

<hr>

</details>

<details>
<summary><b>FAQs for QA Members</b></summary>

<br>

  

  <details>
  <summary><i>There are multiple tests in this issue... Which test should I do first?</i></summary>

  Test in order! Test the first thing first, the second thing second, and so on.

  </details>

  <br>

  

  <details>
  <summary><i>How should I format my issue?</i></summary>

  Here's a template for making issues:

      <b>Test Device</b>

      blah

      <b>Operating System</b>

      blah

      <b>Browser</b>

      blah

      <b>Problem Description</b>

      blah

      <b>Steps to Reproduce</b>

      blah

      <b>Visuals</b>

      blah

      <details>
      <summary><b>Troubleshooting Information</b></summary>

      blah

      </details>

  </details>

  <br>

  

  <details>
  <summary><i>Who should I assign?</i></summary>

  We typically assign the developer who opened the issue in the QA repository.

  </details>

  <br>

  

  <details>
  <summary><i>My question isn't in here... What should I do?</i></summary>

  You should:

  1. Consult the [QA Book](https://github.com/phetsims/QA/blob/master/documentation/qa-book.md).
  2. Google it.
  3. Ask Katie.
  4. Ask a developer.
  5. Google it again.
  6. Cry.

  </details>

<br>

<hr>

</details>

**Comment by samreid** [comment]
A reminder that https://github.com/phetsims/studio/issues/202 is to be included in this version.

**Comment by ariel-phet** [comment]
@samreid you can go ahead and publish

**Comment by samreid** [comment]
Publication is blocked by https://github.com/phetsims/studio/issues/202, on hold until that issue is no longer blocking.

**Comment by samreid** [comment]
https://github.com/phetsims/studio/issues/202 is closed, I'll check in today about whether there are other issues that block publication.  There are 6 issues labeled as such, but I don't know which pertain to this release.

**Comment by arouinfar** [comment]
@samreid now that https://github.com/phetsims/studio/issues/202 has wrapped up, I need to edit the client guides in https://github.com/phetsims/phet-io-client-guides/issues/9.

If you decide to go ahead and build GAO, you'll need to cherry-pick client guide changes.

**Comment by samreid** [comment]
I'd prefer to wait until that issue is addressed, please let me know when that's ready.

**Comment by samreid** [comment]
It seems the last blocking issue is https://github.com/phetsims/sun/issues/676 which is scheduled for developer meeting today.  Perhaps I should wait until after that is discussed to take SHAs.

**Comment by samreid** [comment]
Deployed: https://phet-dev.colorado.edu/html/gravity-and-orbits/1.4.0-rc.1/phet/gravity-and-orbits_en_phet.html
Deployed: https://phet-dev.colorado.edu/html/gravity-and-orbits/1.4.0-rc.1/phet-io/

Leaving self-assigned to update the description with the links.

Assigning to @kathy-phet and @KatieWoe for prioritization.  Would we like this to remain "RC-Lite" testing as we previously discussed?

@arouinfar would you like to do a sanity test before the QA team begins?

**Comment by KatieWoe** [comment]
Per @kathy-phet this is no longer rc lite and is a full rc

**Comment by arouinfar** [comment]
> @arouinfar would you like to do a sanity test before the QA team begins?

Everything looks reasonable to me @samreid

**Comment by jessegreenberg** [comment]
@samreid  @KatieWoe gravity-and-orbits scenery SHA received a patch for phetsims/phet-android-app#155.

If possible, please verify this sim works well in the PhET Android app as part of the next RC test.

**Comment by KatieWoe** [comment]
Phet brand memory test:
<img width="416" alt="GOphetmem" src="https://user-images.githubusercontent.com/41024075/110388298-171cbc00-8020-11eb-8f60-397f776bade0.png">
Phet-io simulation memory test:
<img width="411" alt="GOiomem" src="https://user-images.githubusercontent.com/41024075/110388850-f012ba00-8020-11eb-9443-42f48c378019.png">

**Comment by KatieWoe** [comment]
QA is done

**Comment by samreid** [comment]
Thanks for all the great testing!  I'll continue in https://github.com/phetsims/gravity-and-orbits/issues/374, closing.
